### PR TITLE
adding concourse and mockserver

### PR DESCRIPTION
**Pull request by gauravic08**
Added Concourse under CI and mockserver under Test Framework.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -212,6 +212,7 @@ A curated list of awesome Java frameworks, libraries and software.
 * [Jenkins](http://jenkins-ci.org/) - Provides server-based deployment services.
 * [TeamCity](http://www.jetbrains.com/teamcity/) - JetBrain's CI solution with a free version.
 * [Travis](https://travis-ci.org) - Hosted service often used for open-source projects.
+* [Concourse](https://concourse.ci/) - Provides dockerized pipeline with configuration that can be save in version control.
```
Comment (gauravic08) [inline README.md]: Yes, It's language independent CI/CD tool and uses dockerized environment for build pipeline.

File: README.md [hunk 2]
```diff
@@ -735,7 +736,8 @@ A curated list of awesome Java frameworks, libraries and software.
 * [Selenium](http://docs.seleniumhq.org/) - Portable software testing framework for web applications.
 * [Spock](http://spockframework.github.io/spock/docs/) - JUnit-compatible framework featuring an expressive Groovy-derived specification language.
 * [Truth](https://github.com/google/truth) - Google's assertion and proposition framework.
-* [WireMock](http://wiremock.org/) - Stubbs and mocks web services.
+* [WireMock](http://wiremock.org/) - Stubs and mocks web services.
+* [MockServer](http://www.mock-server.com/) - Mocking framework for Http and Https used for integration testing.
```
Comment (gauravic08) [inline README.md]: Shall I update the definition ?

Comment (akullpp) [inline README.md]: Yes please.

File: README.md (hunk 1 above)
Comment (akullpp) [inline README.md]: Why should we add it then? What about other CI platforms that use containers?

Comment (gauravic08) [inline README.md]: Not sure of any other tool which uses dockerized pipeline and configuration can be saved in version control.

Comment (jbduncan) [inline README.md]: Doesn't Travis have a thing where you run CI jobs inside containers?

I think the unique thing about concourse is it's fully open source and it's not really a platform per sae but a tool which can be integrated into other systems,. AFAIK no other CI platform is like this. But I may be wrong about all this, as I just quickly went over the website and I'm not familiar with how the other CI platforms are developed and managed.

Comment (gauravic08) [inline README.md]: I never used Travis and was considering only open source projects, so suggested Concourse.
Yes, It's a tool which can be integrated into any system.

Comment (akullpp) [inline README.md]: Thanks. That's a great unique selling point for the tool. 😄

**Pull request merged**